interfaces/HO-Portal/src/app/iframe/recipient.page.ts

```diff
@@ -1,51 +1,61 @@
-import { Component, OnInit } from '@angular/core';
-import RegistrationStatus from '../enums/registration-status.enum';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, Params } from '@angular/router';
+import { Subscription } from 'rxjs';
 import { Person } from '../models/person.model';
+import { Program } from '../models/program.model';
+import { ProgramsServiceApiService } from '../services/programs-service-api.service';
 
 @Component({
   selector: 'app-recipient-page',
   templateUrl: './recipient.page.html',
   styleUrls: ['./recipient.page.scss'],
 })
-export class RecipientPage implements OnInit {
+export class RecipientPage implements OnInit, OnDestroy {
   public recipients: Person[];
-  private queryParamPhonenumber = '+15005550002';
+  public programsMap: { [programId: number]: Program };
+  public queryParamPhonenumber = '';
+  public accordionGroupValue = undefined;
+  private paramsSubscription: Subscription;
 
-  constructor() {}
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private progamsServiceApiService: ProgramsServiceApiService,
+  ) {
+    this.paramsSubscription = this.activatedRoute.queryParams.subscribe(
+      (params: Params) => {
+        if (!params.phonenumber) {
+          return;
+        }
+        this.queryParamPhonenumber = params.phonenumber;
```

Comment:
This way, when the query-parameter gets 'emptied' or removed (by Twilio Flex), that won't be reflected in the local/public property `this.queryParamPhonenumber`...